File: day03/part2.swift
var total = 0
var enabled = true
while let line = readLine() {
    let expression = #/do\(\)|don't\(\)|mul\((\d{1,3}),(\d{1,3})\)/#
    for match in line.matches(of: expression) {
        switch match.output.0 {
            case "do()": enabled = true
            case "don't()": enabled = false
            default: if (enabled) { total += Int(match.output.1!)! * Int(match.output.2!)! }
        }
    }
}

print(total)

File: day03/part2.js
let total = 0;
let enabled = true;
for await (const line of console) {
    const expression = new RegExp(`do\\(\\)|don't\\(\\)|mul\\((\\d{1,3}),(\\d{1,3})\\)`, "g");
    for (const [wholeMatch, num1, num2] of line.matchAll(expression)) {
        switch (wholeMatch) {
            case "do()": enabled = true; continue;
            case "don't()": enabled = false; continue;
            default: if (enabled) { total += Number(num1) * Number(num2); }
        }
    }
}

console.log(total);